controller/scheduler-v2/scheduler.go

```diff
@@ -339,7 +337,13 @@ func (s *Scheduler) HandleFormationChange(ef *ct.ExpandedFormation) {
 		s.sendEvent(NewEvent(EventTypeFormationChange, err, nil))
 	}()
 
-	log := logger.New("fn", "HandleFormationChange", "app.id", ef.App.ID, "release.id", ef.Release.ID)
+	log := logger.New("fn", "HandleFormationChange")
+	if ef.App != nil {
+		log = log.New("app.id", ef.App.ID)
+	}
+	if ef.Release != nil {
+		log = log.New("release.id", ef.Release.ID)
+	}
```

Comment:
HandleFormationChange is being called in the scheduler loop, which is supposed to do nothing except pass through the received structs.